packages/controllers/dataset_form.coffee

```diff
@@ -0,0 +1,33 @@
+if Meteor.isClient
+  Template.datasetForm.events
+    'submit form': (event, template) ->
+      event.preventDefault()
+      inputFile = event.target.file?.files[0]
+      unless inputFile
+        toastr.error("Error")
+      else
+        file = new FS.File inputFile
```

Comment:
All of the rest of this logic feels model-level to me -- in my mind, the controller's job is just to collect information from the user (e.g. via a form) and then delegate the creation of records, etc to the models. I'm not sure if Astronomy has class methods built in, but this feels like it should be in a Dataset.createWithRawFile(fields) method that would create a Dataset and a RawFile and could be unit tested with dummy data.